src/CommunityToolkit.Maui/Layouts/DockLayout.shared.cs

```diff
@@ -0,0 +1,65 @@
+﻿using CommunityToolkit.Maui.Core.Interfaces;
+using CommunityToolkit.Maui.Core.Layouts;
+using CommunityToolkit.Maui.Core.Primitives;
+using Microsoft.Maui.Layouts;
+
+namespace CommunityToolkit.Maui.Layouts;
+
+/// <summary>DockLayout positions its child elements along the edges of the layout container.</summary>
+public class DockLayout : Layout, IDockLayout
+{
+	/// <summary>Docking position for a view.</summary>
+	public static readonly BindableProperty DockPositionProperty = BindableProperty.Create(nameof(DockPosition), typeof(DockPosition), typeof(DockLayout), DockPosition.None);
+
+	/// <summary>If true, the last child fills the remaining space. Default value is true.</summary>
+	public static readonly BindableProperty LastChildFillProperty = BindableProperty.Create(nameof(LastChildFill), typeof(bool), typeof(DockLayout), true);
+
+	/// <summary>Horizontal spacing between docked views.</summary>
+	public static readonly BindableProperty HorizontalSpacingProperty = BindableProperty.Create(nameof(HorizontalSpacing), typeof(double), typeof(DockLayout), 0.0);
+
+	/// <summary>Vertical spacing between docked views.</summary>
+	public static readonly BindableProperty VerticalSpacingProperty = BindableProperty.Create(nameof(VerticalSpacing), typeof(double), typeof(DockLayout), 0.0);
+
+	/// <inheritdoc/>
+	public bool LastChildFill
```

Comment:
Maybe name it: ShouldFillLastChild?

Reply:
How about `ExpandLastChild`? `ShouldFillLastChild` sounds very strange. We are not filling the child; we are filling the container space with the child (view).

Reply:
To be more clear for devs to find/know this is a boolean property, we would like to follow a pattern, that's adding a prefix (Is/Has/Should/Will/Do/Must/etc) to the property

e.g.: `IsLastChildFill`

Reply:
Yup! Variable names for Booleans should a question that the value of the Boolean answers, eg IsBusy, IsVisible, etc

Reply:
@brminnick: With this criterium, I could also use `ExpandLastChild`, not? Anyway, what name would you suggest/approve? So that we can resolve this discussion?